src/types.rs

```diff
@@ -128,23 +128,25 @@ impl From<&format::AccessList> for AccessList {
 ///
 /// See ethereum rpc spec for the meaning of fields
 #[napi(object)]
-#[derive(Debug, Default, Clone, PartialEq, Eq)]
+#[derive(Debug, Clone)]
 pub struct Authorization {
-    pub chain_id: String,
-    pub address: String,
-    pub nonce: String,
-    pub y_parity: String,
-    pub r: String,
-    pub s: String,
+    pub chain_id: BigInt, // uint256
+    pub address: String,  // 20-byte hex
+    pub nonce: i64,       // uint64
+    pub y_parity: i64,    // 0 | 1
+    pub r: String,        // 32-byte hex
+    pub s: String,        // 32-byte hex
```

Comment:
Better to put these in doc comments then you see it on lsp hover and stuff. And should be generated into the type declaration file
```suggestion
    /// uint256
    pub chain_id: BigInt,
    /// 20-byte hex
    pub address: String,
    /// uint64
    pub nonce: i64,
    /// 0 | 1
    pub y_parity: i64,
    /// 32-byte hex
    pub r: String,
    /// 32-byte hex        
    pub s: String,
```